### PR TITLE
doc/user/logging.md: update to include a runnable standalone example

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
Update the Logging docs to include a runnable standalone example to demonstrate the `--zap-*` flags.

**Motivation for the change:**
See #1527

### CODE REVIEW

File: doc/user/logging.md [hunk 1]
```diff
@@ -2,29 +2,48 @@
...
-  pflag.CommandLine.AddFlagSet(zap.FlagSet())
-  pflag.Parse()
-  logf.SetLogger(zap.Logger())
-  log := logf.Log.WithName("cmd")
+var globalLog = logf.Log.WithName("global")
 
-  ...
+func main() {
+	pflag.CommandLine.AddFlagSet(zap.FlagSet())
+	pflag.Parse()
```
Comment (estroz) [inline doc/user/logging.md]: Formatting.

File: doc/user/logging.md [hunk 2]
```diff
@@ -63,41 +81,41 @@ import (
...
-        // Include the error in records produced by this log statement.
-        reqLogger.Error(err, "Failed to create new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
-        return reconcile.Result{}, err
-      }
-    }
-    return reconcile.Result{}, err
-  }
-
-  ...
+	// Create a logger for Reconcile() that includes "Request.Namespace"
```
Comment (estroz) [inline doc/user/logging.md]: Formatting.

File: doc/user/logging.md (hunk 1 above)
Comment (estroz) [reply doc/user/logging.md]: Disregard. Looks fine in rendered file.

File: doc/user/logging.md (hunk 2 above)
Comment (estroz) [reply doc/user/logging.md]: Disregard. Looks fine in rendered file.

File: doc/user/logging.md (hunk 1 above)
Comment (joelanford) [reply doc/user/logging.md]: This is just me switching everything from spaces to tabs (since that's what go fmt uses).

**Comment by joelanford** [comment]
@LiliC I think that's a good idea. I'll add another section for that.

**Pull request merged**